packages/x-date-pickers/src/AdapterLuxon/AdapterLuxon.ts

```diff
@@ -259,36 +296,42 @@ export class AdapterLuxon implements MuiPickersAdapter<DateTime, string> {
       return false;
     }
 
-    return this.date(value)!.equals(this.date(comparing)!);
+    return +this.date(value)! === +this.date(comparing)!;
```

Comment:
Technically speaking, this is a breaking change for the Luxon adapter, because now `isEqual` returns true if you have 2 dates with the same timestamp but a different zone.

I think we can consider it to be a bug fix because the other adapters are not working the same way and are already based on the timestamp.